src/compiler/checker.ts

```diff
@@ -9398,13 +9944,21 @@ namespace ts {
         }
 
         function getReturnTypeFromBody(func: FunctionLikeDeclaration, contextualMapper?: TypeMapper): Type {
+            let type: Type;
+            if (func.parserContextFlags & ParserContextFlags.JavaScriptFile) {
+                type = getReturnTypeFromJSDocComment(func);

```

Comment:
I have no problem with this being moved.  It simply seemed like the most natural place to fit it in at the time.  However, looking at it now, i agree it is odd.  THe method is clearly for determing the return type based on the body of the of the function.  And this is not using the body.   So def feel free to move wherever you feel is more appropriate.